code/game/machinery/firealarm.dm

```diff
@@ -326,3 +326,185 @@
 	if (!party_overlay)
 		party_overlay = iconstate2appearance('icons/turf/areas.dmi', "party")
 	A.add_overlay(party_overlay)
+
+/obj/machinery/sprinkler

```

Comment:
Personally I would put this and the refill together in a separate file rather than together with fire alarms